service/history/workflow/update/registry.go

```diff
@@ -219,6 +227,51 @@ func (r *registry) FindOrCreate(ctx context.Context, id string) (*Update, bool,
 	return upd, false, nil
 }
 
+func (r *registry) TryResurrect(ctx context.Context, acptOrRejMsg *protocolpb.Message) (*Update, error) {
+	if acptOrRejMsg == nil || acptOrRejMsg.Body == nil {
+		return nil, nil
+	}
+
+	// Check only total limit here. This might add more than maxInFlight updates to registry,
+	// but provides better developer experience.
+	if err := r.checkTotalLimit(ctx); err != nil {
+		return nil, err
+	}
+
+	body, err := acptOrRejMsg.Body.UnmarshalNew()
+	if err != nil {
+		return nil, invalidArgf("unable to unmarshal request: %v", err)
+	}
+	err = utf8validator.Validate(body, utf8validator.SourceRPCRequest)
+	if err != nil {
+		return nil, invalidArgf("unable to unmarshal request: %v", err)
+	}
+	var reqMsg *updatepb.Request
+	switch updMsg := body.(type) {
+	case *updatepb.Acceptance:
+		reqMsg = updMsg.GetAcceptedRequest()
+	case *updatepb.Rejection:
+		reqMsg = updMsg.GetRejectedRequest()
```

Comment:
Maybe add a Completion case with an error message, even if we think it's impossible?

Would it be appropriate to add a default case which panics, or returns an error, in case in the future someone adds a new branch to the `oneof`?

Reply:
Word `Try` in method name means this method does it best and if it fails then it does nothing. And I believe it is better to ignore future messages too.